### DISCUSSION TITLE
can we make a Go version of the library?

### DISCUSSION THREAD

**Discussion by janeoa** [jfjlaros/arduino-simple-rpc#19] [Ideas]
I already have a ROS node made on Go for the Arduino and was sending JSON thought serial.

Now I want to increase the data rate by switching to the RPC. 
Do you think that it is a good idea to try to implement your python part in Go?

Could I do it in a week or less?

**Comment by jfjlaros** [comment]
Yes, that sounds like a nice idea.

The files `simple_rpc.py`, `protocol.py` and `io.py` form the core of this library. The total line count is about 270, so I guess it should not be too much work to port this to Go. Perhaps the documentation of the [protocol](https://simplerpc.readthedocs.io/en/latest/protocol.html) is also of help to you.

Please let me know if you need any help.

**Reply by janeoa** [reply]
Thanks a lot! ✨
I'm planning to do it in spare time with my friend on https://github.com/janeoa/ardugorpc

lets keep in touch
